lib/replica.rb

```diff
@@ -207,11 +210,17 @@ def compile_revision_ids_query(revisions)
   # These are typical network errors that we expect to encounter.
   TYPICAL_ERRORS = [Errno::ETIMEDOUT, Net::ReadTimeout, Errno::ECONNREFUSED,
                     Oj::ParseError].freeze
-  def report_exception(error, endpoint, query, level='error')
-    level = 'warning' if TYPICAL_ERRORS.include?(error.class)
-    Raven.capture_exception error, level: level, extra: {
-      query: query, endpoint: endpoint, language: @wiki.language, project: @wiki.project
-    }
-    return nil
+
+  def invoke_error_handling_tasks(error, endpoint, query: nil, url: nil, response: nil)
```

Comment:
I think most of this can and should be moved into the error handling module. The closer we get to uniform handling of API errors, instead of treating each different API in a unique way, the easier it will be to extend this strategy to additional places.

There's a lot of duplication between this and OresApi.

Reply:
WikiApi is not using this code but the OresApi and Replica are. So I refactored this into ErrorHandling module and added comment regarding the same.